kasa/discover.py

```diff
@@ -128,6 +169,10 @@ def datagram_received(self, data, addr) -> None:
         device = device_class(ip, port=port)
         device.update_from_discover_info(info)
 
+        # Override protocol if klap created and used succesfully
+        if protocol:
+            device.override_protocol(protocol)
```

Comment:
This should be done in the instance initialization method where needed. Instead of having a separate `override_protocol`, I'd simply set the instance variable for the time being.

Reply:
Ok